spec/abci/abci++_app_requirements.md

```diff
@@ -212,37 +204,37 @@ Nevertheless, as all ABCI calls are now synchronous, ABCI messages using the sam
 still received in sequence.
  -->
 
-#### BeginBlock - DeliverTx - EndBlock
-
-When the consensus algorithm decides on a block, CometBFT uses the sequence of calls to
-`BeginBlock`, `DeliverTx` and `EndBlock` to send the
-decided block's data to the Application, which uses it to transition its state.
+#### FinalizeBlock
 
-The sequence of `DeliverTx` calls is asynchronous but all those calls are enclosed by calls to `BeginBlock` and `EndBlock` which are synchronous.
-
-The Application must remember the latest height from which it
-has run a successful `Commit` so that it can tell CometBFT where to
-pick up from when it recovers from a crash. See information on the Handshake
-[here](#crash-recovery).
+When the consensus algorithm decides on a block, CometBFT uses `FinalizeBlock` to send the
+decided block's data to the Application, which uses it to transition its state, but not persist it;
+persisting will be done during `Commit`.
```

Comment:
```suggestion
decided block's data to the Application, which uses it to transition its state, but MUST NOT persist it;
persisting MUST be done during `Commit`.
```
I'm suggesting this now because this helps in tackling #203